packages/framework-core/src/services/event-store.ts

```diff
@@ -17,90 +17,72 @@ const originOfTime = new Date(0).toISOString() // Unix epoch
+   * are passed, in that case, it won't load the events, but rather it will
+   * proceed with the creation of the new snapshot by considering `pendingEnvelopes`
+   * as the latest events.
+   *
+   * Also, in order to make next calls faster, this method stores the calculated
+   * snapshot.
+   */
   public async fetchEntitySnapshot(entityName: string, entityID: UUID): Promise<EventEnvelope | null> {
     const logger = getLogger(this.config, 'EventStore#fetchEntitySnapshot')
     logger.debug(`Fetching snapshot for entity ${entityName} with ID ${entityID}`)
     const latestSnapshotEnvelope = await this.loadLatestSnapshot(entityName, entityID)
 
-    // eslint-disable-next-line @typescript-eslint/no-extra-parens
     const lastVisitedTime = latestSnapshotEnvelope?.snapshottedEventCreatedAt ?? originOfTime
     const pendingEvents = await this.loadEventStreamSince(entityName, entityID, lastVisitedTime)
 
     if (pendingEvents.length <= 0) {
       return latestSnapshotEnvelope
     } else {
-      logger.debug(
-        `[EventStore#fetchEntitySnapshot] Looking for the reducer for entity ${entityName} with ID ${entityID}`
-      )
+      logger.debug(`Looking for the reducer for entity ${entityName} with ID ${entityID}`)
       let newEntitySnapshot = latestSnapshotEnvelope
       for (const pendingEvent of pendingEvents) {
-        newEntitySnapshot = await this.entityReducer(newEntitySnapshot, pendingEvent)
+        // We double check that what we are reducing is an event
+        if (pendingEvent.kind === 'event') {
+          newEntitySnapshot = await this.entityReducer(newEntitySnapshot, pendingEvent)
+        }
       }
 
-      logger.debug(
-        `[EventStore#fetchEntitySnapshot] Reduced new snapshot for entity ${entityName} with ID ${entityID}: `,
-        newEntitySnapshot
-      )
-
-      return newEntitySnapshot
-    }
-  }
-
-  public async calculateAndStoreEntitySnapshot(
-    entityName: string,
-    entityID: UUID,
-    pendingEnvelopes: Array<EventEnvelope>
-  ): Promise<Array<EventEnvelope>> {
-    const logger = getLogger(this.config, 'EventStore#calculateAndStoreEntitySnapshot')
-    logger.debug('Processing events: ', pendingEnvelopes)
-    logger.debug(`Fetching snapshot for entity ${entityName} with ID ${entityID}`)
-    const latestSnapshotEnvelope = await this.loadLatestSnapshot(entityName, entityID)
-    const snapshotsPerNewEntityId: Record<string, EventEnvelope> = {}
-    if (latestSnapshotEnvelope) {
-      snapshotsPerNewEntityId[latestSnapshotEnvelope.entityID.toString()] = latestSnapshotEnvelope
-    }
+      if (newEntitySnapshot?.entityID !== entityID) {
+        logger.debug(
+          `Migrated entity ${entityName} with previous ID ${entityID} to ${newEntitySnapshot?.typeName} with the new ID ${newEntitySnapshot?.entityID}`,
+          newEntitySnapshot
+        )
+      } else {
+        logger.debug(`Reduced new snapshot for entity ${entityName} with ID ${entityID}: `, newEntitySnapshot)
```

Comment:
What happens if newEntitySnapshot is undefined? Could we say that it was migrated?